File: 345.reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
public class Solution {
    public String reverseVowels(String s) {
        char[] array = s.toCharArray();
        int start = 0;
        int end = array.length -1;
        while (start < end){
            if(!checkvowels(array[start]) && !checkvowels(array[end])){
                start++;
                end--;
            }
            else if(!checkvowels(array[start])){
                start++;
            }
            else if(!checkvowels(array[end])){
                end--;
            }
            else {
                char temp = array[start];
                array[start] = array[end];
                array[end] = temp;
                start++;
                end--;
            }
        }
        return new String(array);
        
        
    }
    
    public boolean checkvowels(Character c){
        if(c == 'a' || c == 'e' || c =='i' || c == 'o' || c== 'u' || c =='A' ||c == 'E' || c=='I' || c == 'O' || c == 'U'){
            return true;
        }
        return false;
    }
}

File: 345.reverse-vowels-of-a-string/reverse_vowels_of_a_string.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = ['a','e','i','o','u','A','E','I','O','U']
        lists = list(s)
        start = 0
        end = len(lists) - 1
        while start < end:
            if lists[start] in vowels and lists[end] in vowels:
                temp = lists[start]
                lists[start] = lists[end]
                lists[end] = temp
                start+=1
                end-=1
            elif lists[start] in vowels and lists[end] not in vowels:
                end-=1
            elif lists[start] not in vowels and lists[end] in vowels:
                start+=1
            else:
                start+=1
                end-=1
        return ''.join(lists)
        
1. use and instead of &&
